### PR TITLE
Refactor polaris-unstyled-link component

### PR DESCRIPTION
**Pull request by vladucu**
#### What's here

- refactors to ES6 class
- template-only components
- angle-bracket syntax

### CODE REVIEW

File: addon/components/polaris-unstyled-link.js [hunk 1]
```diff
@@ -92,25 +80,40 @@ export default Component.extend({
...
+  /** @private */
+  dataPolarisUnstyled = 'true';
+
+  /** @deprecated */
+  dataTestId = null;
+
+  // TODO what/where are these options coming from?!
+  // click: mapEventToAction('onClick', {
+  //   preventDefault: false,
+  //   stopPropagation: true,
```
Comment (vladucu) [inline addon/components/polaris-unstyled-link.js]: @andrewpye when you have a sec wondering if you can recall where are these event handling options coming from?

Comment (andrewpye) [reply addon/components/polaris-unstyled-link.js]: @vladucu apparently came from fixing a [test](https://github.com/smile-io/ember-polaris/commit/ec399e4f91380cefa4ad78ce40b7594157bd851b) 🤷‍♂

Comment (vladucu) [reply addon/components/polaris-unstyled-link.js]: thx

**Comment by vladucu** [comment]
@tomnez can I get some eyes on this pls?

**Pull request merged**